contracts/0.8.9/GateSeal.sol

```diff
@@ -0,0 +1,165 @@
+     */
+    address internal pausable;
+
+    /**
+     * @dev account that can seal the gate,
+     *      intended to be a multisig committee
+     */
+    address internal pauser;
+
+    /**
+     * @dev period in seconds for which the pausable will be paused
+     */
+    uint256 internal pauseDuration;
+
+    /**
+     * @dev unix timestamp past which GateSeal will be unusable 
+     */
+    uint256 internal expiryDate;
+
+
+    constructor(address _initializer) {
+        if (_initializer == address(0)) revert ZeroAddress();
+
+        initializer = _initializer;
+    }
+
+    function getInitializer() external view returns(address) {
+        return initializer;
+    }
+
+    function getPausable() external view returns(address) {
+        return pausable;
+    }
+
+    function getPauser() external view returns(address) {
+        return pauser;
+    }
+
+    function getPauseDuration() external view returns(uint256) {
+        return pauseDuration;
+    }
+
+    function getExpiryDate() external view returns(uint256) {
+        return expiryDate;
+    }
+
+    function isInitialized() external view returns(bool) {
+        return _isInitialized();
+    }
+
+    function isExpired() external view returns(bool) {
+        return _isExpired();
+    }
+
+    /**
+     * @notice set the seal config and expiration
+     * @dev can only be called by the initializer set at construction;
+     * @param _pausable contract to be paused
+     * @param _pauser account that can pause the contract, multisig committee
+     * @param _pauseDuration seconds to pause for
+     * @param _shelfLife period after which GateSeal becomes unusable;
+     *                   starting from the moment this function is called
+     */
+    function initialize(
+        address _pausable,
+        address _pauser,
+        uint256 _pauseDuration,
+        uint256 _shelfLife
+    ) external {
+        if (_isInitialized()) revert AlreadyInitialized();
+        if (msg.sender != initializer) revert NotInitializer();
+        if (_pausable == address(0)) revert ZeroAddress();
+        if (_pauser == address(0)) revert ZeroAddress();
+
+        initializer = address(0);
+
+        pausable = _pausable;
+        pauser = _pauser;
+        pauseDuration = _pauseDuration;
+
+        expiryDate = block.timestamp + _shelfLife;
+
+        emit Initialized(_pausable, _pauser, _pauseDuration, expiryDate);
+    }
+
+    /**
+     * @notice pause the contract and expire this GateSeal
+     * @dev can only be called by the pauser;
+     *      expiring the contract by way of setting `expiryDate` to the past second
+     */
+    function sealGate() external {
+        if (!_isInitialized()) revert YetUninitialized();
+        if (_isExpired()) revert Expired();
+        if (msg.sender != pauser) revert NotPauser();
+
+        expiryDate = block.timestamp - 1;
+
+        IPausableUntil(pausable).pause(pauseDuration);
+
+        emit GateSealed(pausable, pauser, pauseDuration);
```

Comment:
can we self-revoke the role here also?
maybe also add a permissionless self-revoke method if expired